### PR TITLE
Stop leaking state-lock when journal is closed

### PR DESCRIPTION
**Pull request by ggezer**
Fix https://github.com/Alluxio/alluxio/issues/13904

### CODE REVIEW

**Review by jenoudet — APPROVED** [review]

**Review by apc999 — CHANGES_REQUESTED** [review]
Thanks for the quick fix!

Possibly to create a UT to cover this case?
and we can link the UT to the issue for future references

**Review by maobaolong — APPROVED** [review]
LGTM. Thanks for this fix.

File: core/server/master/src/test/java/alluxio/master/journal/JournalContextTest.java [hunk 1]
```diff
@@ -174,6 +174,22 @@ public void pauseBlocksJournalContext() throws Exception {
...
+    // Secondary journals will be closed for operation.
+    mJournalSystem.losePrimacy();
+    // Validate that createJournalContext fails for secondary journals.
+    boolean failedAsExpected = false;
+    try {
+      mBlockMaster.createJournalContext();
+    } catch (UnavailableException e) {
+      failedAsExpected = true;
+    }
+    assertTrue(failedAsExpected);
```
Comment (apc999) [inline core/server/master/src/test/java/alluxio/master/journal/JournalContextTest.java:188]:
```suggestion
    try {
      mBlockMaster.createJournalContext();
      fail("Should have thrown an exception");
    } catch (UnavailableException e) {
      // expected to fail
    }
```

**Review by apc999 — APPROVED** [review]
Minor comments on the test pattern. 
LGTM modulo the comments

File: core/server/master/src/test/java/alluxio/master/journal/JournalContextTest.java [hunk 2]
```diff
@@ -174,6 +174,22 @@ public void pauseBlocksJournalContext() throws Exception {
     }
   }
 
+  @Test
```
Comment (apc999) [inline core/server/master/src/test/java/alluxio/master/journal/JournalContextTest.java:177]: nit: To link test case to the real issue for more context in the future.
```suggestion
  // See https://github.com/Alluxio/alluxio/issues/13904
  @Test
```

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.6 please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.5 please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.64 please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

**Pull request merged**